### PR TITLE
Ensure sound is assigned with right type

### LINKED ISSUE #8172: Scheduled notifications by the recipient's time zone do not work
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.5
  * Firebase SDK version: 7.11.0
  * Installation method: CocoaPods
  * Firebase Component: Messaging

### [REQUIRED] Step 2: Describe the problem

While using Firebase Cloud Messaging, it turned out that notifications sent on a schedule according to the recipient's time zone were not displayed on the device. Test notifications that are sent for a specific fcm token are working. The documentation does not describe how you can test notifications on a schedule, so I had to come up with my own way.
We created a unique Analytic User Property for push notifications only to certain devices. We also added userInfo logging to the methods `-application:didReceiveRemoteNotification:fetchCompletionHandler:`, `-userNotificationCenter: didReceiveNotificationResponse:withCompletionHandler:` и `- userNotificationCenter:willPresentNotification:withCompletionHandler:`.

First, not all devices received notifications: 2 out of 10 tested devices, notifications reached. All tested devices have iOS10+.

Secondly, those devices that received the notification were not shown it.

Thirdly, we found confirmation of the guess that the notification sent by the recipient's time zone comes in the form of a silent notification, based on which a local notification with a delay is created.

Due to logging, we got a userInfo that we could debug with. In our case, it looked like this:
```json
{
    "aps": {
        "content-available": 1
    },
    "custom_url": "<url in string representation>",
    "gcm.message_id": "0000000000000000",
    "gcm.n.e": 1,
    "gcm.notification.badge": 1,
    "gcm.notification.body": "<custom message>",
    "gcm.notification.sound": "default",
    "gcm.notification.sound2": "default",
    "gcms": "gcm.gmsproc.cm",
    "google.c.a.c_id": 0000000000000000000,
    "google.c.a.c_l": "<name of notification>",
    "google.c.a.e": 1,
    "google.c.a.tc": 1,
    "google.c.a.ts": 0000000000,
    "google.c.a.udt": 1,
    "google.c.cm.lt_end": "2021-06-02 14:00:00",
    "google.c.cm.lt_start": "2021-05-28 16:55:00",
    "google.c.sender.id": 000000000000
}
```

When debugging, I found that the local notification is not added to the UNUserNotificationCenter. When calling the method `-addNotificationRequest:withCompletionHandler:` (FIRMessagingContextManagerService.m:186) in completion an error occurs:
Adding notification request failed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection to service on pid 55 named com.apple.usernotifications.usernotificationservice" UserInfo={NSDebugDescription=connection to service on pid 55 named com.apple.usernotifications.usernotificationservice}

It turned out that the problem lies in adding sound to the UNMutableNotificationContent (FIRMessagingContextManagerService.m:165). In userInfo, the value for the key "gcm.notification.sound" is the string "default", and the UNMutableNotificationContent has the `sound` property of the UNNotificationSound type. I.e., the correct sound setting for UNMutableNotificationContent has to be done.

### PR DESCRIPTION
**Pull request by chliangGoogle**
This fix #8172.

### CODE REVIEW

File: FirebaseMessaging/Sources/FIRMessagingContextManagerService.m [hunk 1]
```diff
@@ -194,6 +166,42 @@ + (void)scheduleiOS10LocalNotificationForMessage:(NSDictionary *)message atDate:
...
+  }
+  if ([apsDictionary[kFIRMessagingContextManagerTitleKey] length]) {
+    content.title = apsDictionary[kFIRMessagingContextManagerTitleKey];
+  }
+
+  if (apsDictionary[kFIRMessagingContextManagerSoundKey]) {
+#if !TARGET_OS_WATCH
+    content.sound =
+        [UNNotificationSound soundNamed:apsDictionary[kFIRMessagingContextManagerSoundKey]];
+#endif
```
Comment (maksymmalyhin) [inline FirebaseMessaging/Sources/FIRMessagingContextManagerService.m:190]: nit: Could you please add comments marking which it belongs to for readability here and at other similar conditions, e.g. 
```
#endif // !TARGET_OS_WATCH
```

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingUtilitiesTest.m [hunk 2]
```diff
@@ -69,9 +69,16 @@ - (void)testAppVersionReturnsEmptyStringWhenNotFound {
 
 - (void)testAppIdentifierReturnsExpectedValue {
   NSString *expectedIdentifier = @"com.me.myapp";
+#if TARGET_OS_WATCH
+  expectedIdentifier = @"com.me.myapp.watchkit.watchkitextensions";
+#endif
   [[[_mainBundleMock stub] andReturn:expectedIdentifier] bundleIdentifier];
   NSString *appIdentifier = FIRMessagingAppIdentifier();
+#if TARGET_OS_WATCH
+  XCTAssertEqualObjects(appIdentifier, @"com.me.myapp.watchkit");
```
Comment (maksymmalyhin) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingUtilitiesTest.m:78]: optional: Would it be possible to define a variable for @"com.me.myapp.watchkit" along `expectedIdentifier` to unify the assertion and get rid of the additional `#if TARGET_OS_WATCH`?

**Review by maksymmalyhin — APPROVED** [review]
LGTM, a couple nits.

File: FirebaseMessaging/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# unreleased
```
Comment (paulb777) [inline FirebaseMessaging/CHANGELOG.md:1]: If it merges today or tomorrow, it will be 8.2.0

**Review by paulb777 — APPROVED** [review]
LGTM, when comments are resolved

File: FirebaseMessaging/Sources/FIRMessagingContextManagerService.m (hunk 1 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/FIRMessagingContextManagerService.m]: Done.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingContextManagerServiceTest.m [hunk 4]
```diff
@@ -269,4 +277,85 @@ - (void)mockSchedulingLocalNotifications {
...
+      @"google.c.a.c_l" : @"test local send with sound",
+      @"google.c.a.e" : @1,
+      @"google.c.a.ts" : @1623753000,
+      @"google.c.a.udt" : @1,
+      @"google.c.sender.id" : @449451107265
+    };
+    XCTAssertEqualObjects(content.userInfo, userInfo);
+#endif  // TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_WATCH
+  }
+}
```
[the remark this answers is not in the document]
Comment (chliangGoogle) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingContextManagerServiceTest.m:359]: This is a test file, resolve it.

File: FirebaseMessaging/Sources/FIRMessagingContextManagerService.m [hunk 5]
```diff
@@ -136,62 +136,75 @@ + (BOOL)handleContextManagerLocalTimeMessage:(NSDictionary *)message {
...
-                     kFIRMessagingMessageCodeContextManagerServiceFailedLocalSchedule,
-                     @"Failed scheduling local timezone notification: %@.", error);
-               }
-             }];
+  NSDictionary *userInfo = [self parseDataFromMessage:message];
+  if (userInfo.count) {
+    content.userInfo = userInfo;
   }
+#endif  // TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_WATCH
+  return content;
```
[the remark this answers is not in the document]
Comment (chliangGoogle) [reply FirebaseMessaging/Sources/FIRMessagingContextManagerService.m:207]: Pretty sure this is tested. mark all for resolved. Not sure why this actions inaccurately pick up the lines and test files.

File: FirebaseMessaging/CHANGELOG.md (hunk 3 above)
Comment (chliangGoogle) [reply FirebaseMessaging/CHANGELOG.md]: Done.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingUtilitiesTest.m (hunk 2 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingUtilitiesTest.m]: Done.

**Pull request merged**